src/raft.c

```diff
@@ -639,35 +640,57 @@ static void clearClientSessions(RedisRaftCtx *rr)
 
 static void timeoutBlockedCommand(RedisRaftCtx *rr, raft_entry_t *entry)
 {
+    RaftReq *req = entryDetachRaftReq(rr, entry);
+
     raft_index_t idx;
-    bool error; /* not used yet */
+    bool error;
 
     int ret = RaftRedisDeserializeTimeout(entry->data, entry->data_len, &idx, &error);
     RedisModule_Assert(ret == RR_OK);
 
     BlockedCommand *bc = getBlockedCommand(idx);
     if (!bc) {
         /* unblock handler called before timeout was applied */
+        RedisModule_ReplyWithLongLong(req->ctx, 0);
```

Comment:
If I'm not mistaken, we won't have req on replay? A test would be nice for these race conditions if possible.

Reply:
yes, it needs to be protected with an if (req) {}  (this is the client unblock req)

Reply:
this test would be to client unblock, and then restart the node to test that replay works as expected and doesn't crash.  (possibly to push also after restart and validate that it didn't get popped)

Reply:
I'm actually unsure how to test get this race condition to work. i.e. to have the blocked command execute before the timeout.  

Only way I can say forcing it is via a debug command that issues 2 entries in succession (basically atomically to redis), first entry is an lpush, second entry is the timeout log entry.  since redis couldn't process them (apply) in between each other, its the closest to simulating the race condition.

Reply:
> I'm actually unsure how to test get this race condition to work. i.e. to have the blocked command execute before the timeout.

Maybe you can use 3 seconds timeout for the blocking command and then submit a lpush from another client

Reply:
How about 
- blpop 3 seconds
- config set raft.log-disable-apply yes
- lpush
- wait at least 3 seconds so timeout entry will be appended
- config set raft.log-disable-apply no

Reply:
oh, didnt know you could do that.  disable apply that is :)  that would work, basically the same as what I pushed already.  perhaps a one off debug command isn't what we want, so can move to your approach.